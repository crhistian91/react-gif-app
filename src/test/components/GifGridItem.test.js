import React from 'react'

import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";


describe('pruebas en <GifGridItem />', () => {

    const title = 'un titulo';
    const url = 'http://google.com';
    const wrapper = shallow( <GifGridItem title={ title } url={ url } /> )

    test('debe de mostrar el componente correctamente', () => {

        expect( wrapper ).toMatchSnapshot();

    })

    test('debe de contener un parrafo con el titulo', () => {
        const img = wrapper.find('img');
        expect( img.prop('src').trim()).toBe(url);
        expect( img.prop('alt').trim()).toBe(title);
    })

    test('la clase debe contener animate__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes('animate__fadeIn')).toBe(true);
    })

})