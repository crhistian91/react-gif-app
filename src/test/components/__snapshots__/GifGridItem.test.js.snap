// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGridItem /> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="un titulo"
    src="http://google.com"
  />
  <p>
     
    un titulo
     
  </p>
</div>
`;
